src/ExpressionEvaluator/Core/Source/ExpressionCompiler/MetadataUtilities.cs

```diff
@@ -135,7 +136,12 @@ private static PortableExecutableReference MakeAssemblyMetadata(ModuleMetadata m
                         // or if the module has not been loaded yet. The value will be null
                         // if the name was ambiguous.
                         ModuleMetadata module;
-                        if (modulesByName.TryGetValue(name, out module) && (module != null))
+                        if (!modulesByName.TryGetValue(name, out module))
+                        {
+                            // AssemblyFile names may contain file information (".dll", etc).
+                            modulesByName.TryGetValue(Path.GetFileNameWithoutExtension(name), out module);
```

Comment:
It's theoretically possible, but I've been assured several times that "A.B.C" should be loaded if we're trying to resolve an address in that module (it's not possible for the debugger to create a DkmClrInstructionAddress to hand to the ee without loading the module).